src/main/java/org/squiddev/plethora/integration/xnet/MethodsController.java

```diff
@@ -0,0 +1,152 @@
+package org.squiddev.plethora.integration.xnet;
+
+
+import mcjty.xnet.XNet;
+import mcjty.xnet.api.channels.IControllerContext;
+import mcjty.xnet.api.keys.SidedPos;
+import mcjty.xnet.blocks.cables.ConnectorTileEntity;
+import mcjty.xnet.blocks.controller.TileEntityController;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.squiddev.plethora.api.IWorldLocation;
+import org.squiddev.plethora.api.method.ContextKeys;
+import org.squiddev.plethora.api.method.IContext;
+import org.squiddev.plethora.api.method.LuaList;
+import org.squiddev.plethora.api.method.wrapper.PlethoraMethod;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+// Based heavily off of the OC XNet Driver mod https://minecraft.curseforge.com/projects/oc-xnet-driver
+public final class MethodsController {
+	private MethodsController() {
+	}
+
+	/*
+	 *Links for convenience in pulling up references:
+	 * https://raw.githubusercontent.com/thraaawn/OCXNetDriver/master/src/main/java/org/dave/ocxnetdriver/driver/controller/EnvironmentXnetController.java
+	 * https://github.com/thraaawn/OCXNetDriver/raw/master/src/main/java/org/dave/ocxnetdriver/converter/ConverterBlockPos.java
+	 *
+	 */
+
+	/*
+	 *Methods defined in the OC XNet Driver:
+	 * getConnectedBlocks
+	 * getSupportedCapabilities(pos[, side])
+	 * getItems(pos[, side])
+	 * getFluids(pos[, side])
+	 * getEnergy(pos[, side])
+	 * transferItem(sourcePos, sourceSlot, amount, targetPos[, sourceSide[, targetSide]])
+	 * transferFluids(sourcePos, amount, targetPos[, fluidName][, sourceSide[, targetSide]])
+	 * transferEnergy(sourcePos, amount, targetPos[, sourceSide[, targetSide]])
+	 * store(sourcePos, sourceSlot, database, entry[, sourceSide]) - Stores an itemstack in an OC database upgrade
+	 *
+	 *Additional data that we may expose, either method or meta
+	 * NetworkId
+	 * Channels
+	 * RF storage - should already be exposed
+	 * Logic state - the colors are defined in mcjty.xnet.api.channels.Color,
+	 *   usage shown in mcjty.xnet.api.helper.AbstractConnectorSettings.calculateColorsMask
+	 *   Alternatively, a player could set a Redstone Proxy and read the state;
+	 *   we could theoretically expose a
+	 */
+
+
+	//----------
+	//Start OC XNet Driver stubs
+	//----------
+	// This block is intended to simplify adapting the code from the OC XNet Driver mod,
+	// and may be removed at a later time.
+
+	//TODO Do we need this?  This was a configuration setting in OC XNet Driver
+	private static final boolean relativePositions = false;
```

Comment:
Always relative! I try to avoid leaking absolute block positions to the user as a) it's a bit of a 4th wall break and b) that's what GPS is for :).

Reply:
Alright, I've removed that boolean and clarified the usage of `toAbsolute` and `toRelative`.